.github/workflows/CI.yml

```diff
@@ -39,6 +39,116 @@ jobs:
         run: npx envinfo
       - name: Build
         run: ./configure && make -j8
+  build-tarball:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Environment Information
+        run: npx envinfo
+      - name: Make tarball
+        run: | 
+          export DISTTYPE=nightly
+          export DATESTRING=`date "+%Y-%m-%d"`
+          export COMMIT=xxxx
+          ./configure && make tar -j8
+          mkdir tarballs
+          mv *.tar.gz tarballs
+      - name: Upload tarball artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: tarballs
+          path: tarballs
+  test-tarball-linux:
+    needs: build-tarball
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Environment Information
+        run: npx envinfo
+      - name: Download tarball
+        uses: actions/download-artifact@v1
+        with:
+          name: tarballs
+      - name: Extract tarball
+        run: tar xzf tarballs/*.tar.gz
+      - name: Build
+        run: |
+          export TAR_DIR=`basename tarballs/*.tar.gz .tar.gz`
+          cd $TAR_DIR
+          ./configure && make -j2
+          cd ..
+          mkdir -p out/Release
+          cp $TAR_DIR/out/Release/node out/Release
+          cp $TAR_DIR/out/Release/cctest out/Release
```

Comment:
I think this can be avoided by passing `--shell` to the test runner:
https://github.com/nodejs/node/blob/c1b2f6afbe03841a6f0ca5fa363ceb5bcdcfe638/tools/test.py#L1345